underfs/s3a/src/main/java/alluxio/underfs/s3a/S3AInputStream.java

```diff
@@ -123,56 +126,40 @@ public long skip(long n) throws IOException {
     if (n <= 0) {
       return 0;
     }
-    closeStream();
-    mPos += n;
-    openStream();
-    return n;
+    long skipped = in.skip(n);
+    mPos += skipped;
+    return skipped;
   }
 
-  /**
-   * Opens a new stream at mPos if the wrapped stream mIn is null.
-   */
-  private void openStream() throws IOException {
-    if (mIn != null) { // stream is already open
+  @Override
+  public long getPos() {
+    return mPos;
+  }
+
+  @Override
+  public void seek(long pos) throws IOException {
+    if (pos == mPos) {
       return;
     }
-    GetObjectRequest getReq = new GetObjectRequest(mBucketName, mKey);
-    // If the position is 0, setting range is redundant and causes an error if the file is 0 length
-    if (mPos > 0) {
-      getReq.setRange(mPos);
+    if (pos < mPos) {
+      mPos = pos;
+      reopenStream();
```

Comment:
line 144-145 needs to be after 146 I think otherwise there would be a condition that reopenStream failed but mPos is already equal to pos. And if user tries to seek again, this function will return on line 142 and cause an issue.

Reply:
Good call, change to reopenStream with new position and update position afterwards